### ISSUE TITLE
PPA for Ubuntu

### ISSUE THREAD

**Issue by ploum** [Zettlr/Zettlr#361]
Is there any plan to have a PPA for Ubuntu? That would be awesome

**Comment by ploum** [comment]
If I remember correctly, it's basically a server where your upload the source of your deb package. The server will then compile it (even for different architecture). 

After that, it can be added to any Ubuntu-based system so Zettlr is now available, with all updates, directly to every user.

If you don't have access to a Linux computer, I may have a look (but I was really bad at packaging)

**Comment by nathanlesage** [comment]
:D That would make sense. If I see this correctly, it all seems to boil down to a certain directory structure (so there's no need with all that overhead hassle). Because what I would like to have is an easy way to automate that so that I only need to re-use my command for updating the API after a release and have everything else done automatically. Anything else I can't manage!

**Comment by ploum** [comment]
Afaik, once the setup is done, uploading a new version to a PPA is only a matter of running the commande "dput". So easily automatable.

What I'm not sure is if it requires an Ubuntu computer.

(well, it looks like this can be uploaded through sftp : https://help.launchpad.net/Packaging/PPA/Uploading )

**Comment by nathanlesage** [comment]
Well, my server runs Debian, so what I hope for is that I can simply automatically pull in the *.deb-file and throw it on the server!

**Comment by almet** [comment]
Heya, anything we can do to help you on this? Thanks!

**Comment by nathanlesage** [comment]
Certainly, if someone would be able to maintain this, it would ease my workload a lot (that is: keep such a repository updated)

**Comment by ScottishJon** [comment]
I don't have the know-how for this, but my understanding of Flatpak as an option for Linux distribution is that it automatically pulls releases from Github once an app has been submitted and set up with them, which I imagine would be less work that setting up and maintaining a PPA for only Debian-based distributions. [Info here](https://github.com/flathub/flathub/wiki/App-Submission).

**Comment by nathanlesage** [comment]
Ugh, looks like quite some work to maintain. Especially the "don't download additional data during build" part got me puzzled, b/c without it, the app won't build

**Comment by ploum** [comment]
I belive this one would still be a highly welcomed contribution. This contribution should come from someone using Ubuntu and knowledgeable with packaging. (I tried it myself several times, I'm definitlely not a good packager).

Maybe a "contribution welcome" or "patch or it won't happen" tag would do it?

**Comment by nathanlesage** [comment]
> (I tried it myself several times, I'm definitlely not a good packager).

My thought exactly! I'll just label this with "help wanted", I think this might be the easiest thing, right?

**Comment by Kangie** [comment]
@nathanlesage I've got a bit of experience packaging for `.debs` so I'll take a look into this after 1.8.0.

I'm going to target Debian, because working packages on Debian, in my experience, install and work just fine on Ubuntu. It should be available to a wider userbase this way! 

As far doing things otherwise goes, I'd probably aim at getting things into the official repos, rather than a PPA. Once things hit the Debian repos (as I understand it) they usually make their way into Ubuntu anyway.

If you'd prefer another approach, let me know.

**Comment by ploum** [comment]
@Kangie > that would clearly be the best approach. But the most time consuming. The disadvantage of this approach is that the version in Ubuntu is "frozen" with each release. Which means disabling the automatic check for a new version.

In a perfect world, there would be a packaged version in Debian and a PPA.

**Comment by nathanlesage** [comment]
> As far doing things otherwise goes, I'd probably aim at getting things into the official repos, rather than a PPA.

This sounds preferable, and I would really appreciate it if we could make it! I can help out with server resources and some aid, but as mentioned, not the fundamental coding work. But if there's a server necessity, I'd be happy to help out here!

**Comment by marado** [comment]
> As far doing things otherwise goes, I'd probably aim at getting things into the official repos, rather than a PPA. Once things hit the Debian repos (as I understand it) they usually make their way into Ubuntu anyway.

You might want to tackle #540 if you wish to get the debs into Debian's official repositories.

**Comment by marado** [comment]
Yes:
```
$ lintian ./Zettlr-2.2.1-amd64.deb |grep copy
E: zettlr: no-copyright-file
```

**Comment by nathanlesage** [comment]
Ahh damn. @kyaso since you have a Linux VM (I'm really sorry that I'm always asking you in these cases! 🙈 ) would you mind having a look at this? After running a release then you could see when this command above doesn't spit out an error anymore …

**Comment by nathanlesage** [comment]
For Christ's sake ... but alright, this issue here is pinned, so it's going to stay open until we're able to fix this!

**Comment by cirosantilli** [comment]
Or `snap`!

**Comment by nathanlesage** [comment]
Hello,

I have now set up a barebones PPA using this guide: https://assafmo.github.io/2019/05/02/ppa-repo-hosted-on-github.html

You can find it here: https://ppa.zettlr.com/

Someone has already reported potential issues with it, and it's really just barebones, so if anyone has tips on how to improve the repository, please let me know, I don't have a Debian laptop for daily use so debugging this is a bit iffy for me.

**Comment by nathanlesage** [comment]
Another senior issue solved — see #5733, we now have a PPA that apparently works well!

Sorry for the delay, but on the bright side, my dissertation is almost done :D
